src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+} from "lit-element";
+
+import { navigate } from "../../../../../common/navigate";
+import { fetchDevices, ZHADevice } from "../../../../../data/zha";
+import "../../../../../layouts/hass-error-screen";
+import "../../../../../layouts/hass-subpage";
+import type { HomeAssistant } from "../../../../../types";
+import { Network, Edge, Node, EdgeOptions } from "vis-network";
+import "vis-network/styles/vis-network.css";
+
+@customElement("zha-network-visualization-page")
+export class ZHANetworkVisualizationPage extends LitElement {
+  @property({ type: Object }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow!: boolean;
+
+  @query("#visualization", true)
+  private _visualization!: HTMLElement;
+
+  @internalProperty()
+  private _devices: Map<string, ZHADevice> = new Map();
+
+  @internalProperty()
+  private _network!: Network;
+
+  protected firstUpdated(changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+    if (this.hass) {
+      this._fetchData();
+    }
+    this._network = new Network(
```

Comment:
You should make sure we are connected when creating this, as otherwise the height and width could not be set yet.

Reply:
Or you should update it in onConnected when `this._network` is defined